dbt/adapters/athena/impl.py

```diff
@@ -46,6 +46,13 @@ def convert_number_type(cls, agate_table: agate.Table, col_idx: int) -> str:
     def convert_datetime_type(cls, agate_table: agate.Table, col_idx: int) -> str:
         return "timestamp"
 
+    def drop_relation(self, relation):
+        if relation.type is None:
+            raise_compiler_error(f"Tried to drop relation {relation}, but its type is null.")
+
+        self.cache_dropped(relation)
```

Comment:
I think this is a good idea and if it works, it is perfect !
A thing I would try is to keep the drop_relation in python but try to rename the macro `athena__drop_relation` and try calling `adapter.drop_relation` in SQL macros.
This way, contributors can use the expected `adapter.drop_relation` and our login enables to :
- Call the right macro 
- **WITH** the config